app/controllers/reporting/reports_controller.rb

```diff
@@ -0,0 +1,27 @@
+module Reporting
+  class ReportsController < ReportingController
+    before_action :set_report_type, except: :index
+    before_action :set_report, except: :index
+
+    before_action :require_dashboard_access!, only: [:index]
+    before_action :require_report_access!, only: [:show]
+
+    def index; end
+
+    def show; end
+
+    private
+
+    def set_report
+      @report = report_type_report_mapping[@report_type]
+    end
+
+    def report_type_report_mapping
```

Comment:
Why not just specify the class directly as before - what is the benefit of the hash lookup - you still have to pass a string specifying the report in the CrimeApplicationController so are we not just doing the same thing with more steps?